drivers/led/is31fl3216a.c

```diff
@@ -0,0 +1,248 @@
+{
+	return is31fl3216a_write_reg(i2c, IS31FL3216A_REG_UPDATE, 0);
+}
+
+static uint8_t is31fl3216a_brightness_to_pwm(uint8_t brightness)
+{
+	return (0xFFU * brightness) / 100;
+}
+
+static int is31fl3216a_led_write_channels(const struct device *dev,
+					  uint32_t start_channel,
+					  uint32_t num_channels,
+					  const uint8_t *buf)
+{
+	const struct is31fl3216a_cfg *config = dev->config;
+	uint8_t i2c_buffer[IS31FL3216A_MAX_LEDS + 1];
+	int status;
+	int i;
+
+	if (num_channels == 0) {
+		return 0;
+	}
+
+	if (start_channel > IS31FL3216A_MAX_LEDS ||
+	    start_channel + num_channels > IS31FL3216A_MAX_LEDS) {
+		return -EINVAL;
+	}
+
+	/* Invert channels, last register is channel 0 */
+	i2c_buffer[0] =
+		IS31FL3216A_REG_PWM_LAST - start_channel - (num_channels - 1);
+	for (i = 0; i < num_channels; i++) {
+		if (buf[num_channels - i - 1] > 100) {
+			return -EINVAL;
+		}
+		i2c_buffer[i + 1] = is31fl3216a_brightness_to_pwm(
+			buf[num_channels - i - 1]);
+	}
+
+	status = is31fl3216a_write_buffer(&config->i2c, i2c_buffer,
+					  num_channels + 1);
+	if (status) {
+		return status;
+	}
+
+	return is31fl3216a_update_pwm(&config->i2c);
+}
+
+static int is31fl3216a_led_set_brightness(const struct device *dev,
+					  uint32_t led, uint8_t value)
+{
+	const struct is31fl3216a_cfg *config = dev->config;
+	uint8_t pwm_reg = IS31FL3216A_REG_PWM_LAST - led;
+	int status;
+	uint8_t pwm_value;
+
+	if (led > IS31FL3216A_MAX_LEDS - 1) {
+		return -ENODEV;
+	}
+
+	if (value > 100) {
+		return -EINVAL;
+	}
+
+	pwm_value = is31fl3216a_brightness_to_pwm(value);
+	status = is31fl3216a_write_reg(&config->i2c, pwm_reg, pwm_value);
+	if (status) {
+		return status;
+	}
+
+	return is31fl3216a_update_pwm(&config->i2c);
+}
+
+static int is31fl3216a_led_on(const struct device *dev, uint32_t led)
+{
+	return is31fl3216a_led_set_brightness(dev, led, 100);
+}
+
+static int is31fl3216a_led_off(const struct device *dev, uint32_t led)
+{
+	return is31fl3216a_led_set_brightness(dev, led, 0);
+}
+
+static int is31fl3216a_init_registers(const struct i2c_dt_spec *i2c)
+{
+	int i;
+	int status;
+
+	status = is31fl3216a_write_reg(i2c, IS31FL3216A_REG_CTL_1, 0xFF);
+	if (status) {
+		return status;
+	}
+
+	status = is31fl3216a_write_reg(i2c, IS31FL3216A_REG_CTL_2, 0xFF);
+	if (status) {
+		return status;
+	}
+
+	status = is31fl3216a_write_reg(i2c, IS31FL3216A_REG_LIGHT_EFFECT, 0x00);
+	if (status) {
+		return status;
+	}
+
+	status = is31fl3216a_write_reg(i2c, IS31FL3216A_REG_CHANNEL_CONFIG,
+				       0x00);
+	if (status) {
+		return status;
+	}
+
+	status = is31fl3216a_write_reg(i2c, IS31FL3216A_REG_GPIO_CONFIG, 0x00);
+	if (status) {
+		return status;
+	}
+
+	status = is31fl3216a_write_reg(i2c, IS31FL3216A_REG_OUTPUT_PORT, 0x00);
+	if (status) {
+		return status;
+	}
+
+	status = is31fl3216a_write_reg(i2c, IS31FL3216A_REG_INT_CONTROL, 0x00);
+	if (status) {
+		return status;
+	}
+
+	status = is31fl3216a_write_reg(i2c, IS31FL3216A_REG_ADC_SAMPLE_RATE,
+				       0x00);
```

Comment:
same as bellow.

Reply:
Unfortunately not but I could fix some others